tests/app/data_model/test_questionnaire_store.py

```diff
@@ -1,185 +1,124 @@
-            expected_completed_block_ids,
-        )
-
-    def test_questionnaire_store_missing_keys(self):
-        # Given
-        expected = get_basic_input()
-        del expected["PROGRESS"]
-        self.input_data = json_dumps(expected)
-        # When
-        store = QuestionnaireStore(self.storage)
-        # Then
-        self.assertEqual(store.metadata.copy(), expected["METADATA"])
-        self.assertEqual(store.response_metadata, expected["RESPONSE_METADATA"])
-        self.assertEqual(store.answer_store, AnswerStore(expected["ANSWERS"]))
-        self.assertEqual(store.progress_store.serialize(), [])
-
-    def test_questionnaire_store_updates_storage(self):
-        # Given
-        expected = get_basic_input()
-        store = QuestionnaireStore(self.storage)
-        store.set_metadata(expected["METADATA"])
-        store.answer_store = AnswerStore(expected["ANSWERS"])
-        store.response_metadata = expected["RESPONSE_METADATA"]
-        store.progress_store = ProgressStore(expected["PROGRESS"])
-
-        # When
-        store.save()  # See setUp - populates self.output_data
-
-        # Then
-        self.assertEqual(expected, json_loads(self.output_data))
-
-    def test_questionnaire_store_errors_on_invalid_object(self):
-        # Given
-        class NotSerializable:
-            pass
-
-        non_serializable_metadata = {"test": NotSerializable()}
-
-        expected = get_basic_input()
-        store = QuestionnaireStore(self.storage)
-        store.set_metadata(non_serializable_metadata)
-        store.response_metadata = expected["RESPONSE_METADATA"]
-        store.answer_store = AnswerStore(expected["ANSWERS"])
-        store.progress_store = ProgressStore(expected["PROGRESS"])
-
-        # When / Then
-        self.assertRaises(TypeError, store.save)
-
-    def test_questionnaire_store_deletes(self):
-        # Given
-        expected = get_basic_input()
-        store = QuestionnaireStore(self.storage)
-        store.set_metadata(expected["METADATA"])
-        store.response_metadata = expected["RESPONSE_METADATA"]
-        store.answer_store = AnswerStore(expected["ANSWERS"])
-        store.progress_store = ProgressStore(expected["PROGRESS"])
-
-        # When
-        store.delete()  # See setUp - populates self.output_data
-
-        # Then
-        self.assertNotIn("a-test-section", store.progress_store)
-        self.assertEqual(store.metadata.copy(), {})
-        self.assertEqual(len(store.answer_store), 0)
-        self.assertEqual(store.response_metadata, {})
-
-    def test_questionnaire_store_raises_when_writing_to_metadata(self):
-        store = QuestionnaireStore(self.storage)
-
-        with self.assertRaises(TypeError):
-            store.metadata["no"] = "writing"
+def test_questionnaire_store_json_loads(questionnaire_store, basic_input):
+    # Given
+    questionnaire_store.input_data = json_dumps(basic_input)
+    # When
+    store = QuestionnaireStore(questionnaire_store.storage)
```

Comment:
Worth considering if this can be extracted to conftest, on a quick check I found 2 test that didn't work afterwards, but might be made to

Reply:
Same issue here, reusable fixture would be right but the changes would be to so many different out of scope tests